src/util/util.js

```diff
@@ -61,10 +62,70 @@ function resolveURLTemplates(URLTemplates, variables = {}, options = {}) {
       resolveURL = resolveURL.replace(macro1, value);
       resolveURL = resolveURL.replace(macro2, value);
     }
-    URLs.push(resolveURL);
+    resolvedURLs.push(resolveURL);
   }
 
-  return URLs;
+  return resolvedURLs;
+}
+
+/**
+ * Extract the url/s from the URLTemplates.
+ *   If the URLTemplates is an array of urls
+ *   If the URLTemplates object has a url property
+ *   If the URLTemplates is a single string
+ *
+ * @param {Array|String} URLTemplates - An array|string of url templates.
+ */
+function extractURLsFromTemplates(URLTemplates) {
+  if (Array.isArray(URLTemplates)) {
+    return URLTemplates.map(URLTemplate => {
+      return URLTemplate && URLTemplate.hasOwnProperty('url')
+        ? URLTemplate.url
+        : URLTemplate;
+    });
+  }
+  return URLTemplates;
+}
+
+/**
+ * Returns a boolean after checking if the object exists in the array.
+ *   true - if the object exists, false otherwise
+ *
+ * @param {Object} obj - The object who existence is to be checked.
+ * @param {Array} list - List of objects.
+ */
+function containsObject(obj, list) {
```

Comment:
sorry just another thing to add, the names of these functions shouldn't be so generic because in fact this function is only usable for url templates. same for `joinArrayOfUniqueObjs`.

Reply:
Updated to reflect functionality